plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryStorageArrowPageSource.java

```diff
@@ -92,13 +100,14 @@ public long getReadTimeNanos()
     @Override
     public boolean isFinished()
```

Comment:
The issue which these changes are supposed to fix had all the threads stuck inside this method
```
"20240829_051047_04581_5tn6d.3.4.0-26-98" #98 [919667] prio=5 os_prio=0 cpu=145595.11ms elapsed=54281.20s tid=0x00007f7d7af90df0 nid=919667 waiting on condition  [0x00007f48eaceb000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@22.0.2/Native Method)
	- parking to wait for  <0x00007f4fb7dfdab8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(java.base@22.0.2/LockSupport.java:371)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(java.base@22.0.2/AbstractQueuedSynchronizer.java:519)
	at java.util.concurrent.ForkJoinPool.unmanagedBlock(java.base@22.0.2/ForkJoinPool.java:4011)
	at java.util.concurrent.ForkJoinPool.managedBlock(java.base@22.0.2/ForkJoinPool.java:3959)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@22.0.2/AbstractQueuedSynchronizer.java:1707)
	at java.util.concurrent.ArrayBlockingQueue.take(java.base@22.0.2/ArrayBlockingQueue.java:420)
	at com.google.api.gax.rpc.QueuingResponseObserver.getNext(QueuingResponseObserver.java:70)
	at com.google.api.gax.rpc.ServerStreamIterator.hasNext(ServerStreamIterator.java:96)
	at io.trino.plugin.bigquery.ReadRowsHelper$ReadRowsIterator.hasNext(ReadRowsHelper.java:79)
	at io.trino.plugin.bigquery.BigQueryStorageArrowPageSource.isFinished(BigQueryStorageArrowPageSource.java:95)
	at io.trino.operator.ScanFilterAndProjectOperator$ConnectorPageSourceToPages.process(ScanFilterAndProjectOperator.java:376)
```

Looking at `com.google.api.gax.rpc.ServerStreamIterator#isReady` it seems any call to hasNext or next can block. So we should use futures for `next` calls rather than for the creation of the iterator and `isFinished` should be changed to `responses.isReady() && !responses.hasNext()`